src/app/shipping/SingleShippingForm.tsx

```diff
@@ -89,6 +89,14 @@ class SingleShippingForm extends PureComponent<SingleShippingFormProps & WithLan
         }, SHIPPING_AUTOSAVE_DELAY);
     }
 
+    componentDidMount(): void {
+        const { consignments } = this.props;
+
+        if ( consignments.length > 1) {
+            this.handleAddressSelect(consignments[0].shippingAddress);
```

Comment:
Ok I did some digging and it seems SDK does some magic where calling `CheckoutService#updateAddress` would [re-assign](https://github.com/bigcommerce/checkout-sdk-js/blob/master/src/shipping/consignment-action-creator.ts#L156) all line items to the consignment of the address.

Could we add some unit tests for this? It is a fix so it means you're fixing a scenario not covered by our existing tests.

Reply:
Sure, I'll add unit test(s) and run functional tests on Launchbay.

Reply:
Sorry @bc-peng, I had a look at this again and realised that perhaps a better solution is to make a call to combine the consignments when a user clicks on the "Ship to single address" button. The reason is because we're making a destructive call when this form component is mounted. I'm concerned that it's likely to cause problems in the future because engineers don't usually expect things to be deleted/updated just for showing something. But if the delete/update is explicitly tied to a user action, it's going to be easier to reason about. I know functionally speaking,  the two options have the same outcome because the form component is only mounted when a shopper clicks on the "Ship to single address" button. But latter option is more explicit. What do you think about this?

Reply:
Hi @davidchin , the first thing I did in approaching this ticket was to locate [the function](https://github.com/bigcommerce/checkout-js/blob/84d758cbbd71be98a3c307bd0f34fc941248779a/src/app/checkout/Checkout.tsx#L465) triggered by the `Ship to single address` button. It was in the very high level `Checkout.tsx`.  [The API call](https://github.com/bigcommerce/checkout-js/blob/84d758cbbd71be98a3c307bd0f34fc941248779a/src/app/shipping/SingleShippingForm.tsx#L224) was in `singleShippingForm.tsx`, far away from `Checkout.tsx`.

I didn't figure out a straightforward way to trigger an API request from `Checkout.tsx`. Hence I had chosen to put everything inside `singleShippingForm.tsx`.

Reply:
Update: It would make sense to add consignment number check to the existing single shipping handle function🤦🏻‍♂️,, `handleSingleShippingSubmit`, [Shipping.tsx#L171](https://github.com/bigcommerce/checkout-js/blob/84d758cbbd71be98a3c307bd0f34fc941248779a/src/app/shipping/Shipping.tsx#L171).

I have submitted the new code based on this idea.